stack/documentation-system-overview.rst

```diff
@@ -0,0 +1,179 @@
+===========================================
+
+This section is a high-level overview of the tools and processes for building Stack documentation.
+To learn how to build the `pipelines.lsst.io`_ site in practice, see :doc:`building-pipelines-lsst-io-locally`.
+
+.. _stack-docs-system-build-documenteer:
+
+Documenteer
+-----------
+
+Documenteer_ is the Python package that provides tooling for LSST DM’s Sphinx-based documentation projects, including Stack user guides such as `pipelines.lsst.io`_.
+Documenteer_ plays four roles:
+
+- Documenteer_ manages and versions the Python dependencies needed to build Sphinx documentation.
+  The :file:`requirements.txt` file in the `pipelines_lsst_io`_ repository refers to a specific release of Documenteer_, which in turn specifies its own Python dependencies.
+
+- Documenteer_ provides centralized Sphinx_ configurations for both the `pipelines_lsst_io`_ repository and individual packages.
+
+- Documenteer_ provides Sphinx_ extensions built specifically for LSST DM needs.
+
+- Documenteer provides the `stack-docs`_ command-line app that runs the Sphinx_ documentation build process (Documenteer_ also provides the related `package-docs`_ app for :doc:`single-package documentation builds <building-single-package-docs>`).
+
+.. _stack-docs-system-build-setup:
+
+Stack build and set up
+----------------------
+
+The prerequisite for the Stack documentation build is that the Stack is already built and set up.
+Not only does this make the Stack importable so that Sphinx can introspect the Python APIs, it also means that Doxygen has already run (through SCons_\ /`sconsUtils`_ and `lsstDoxygen`_) and generated XML files with C++ API descriptions.
+
+The :ref:`main documentation repository <stack-docs-system-main-repo>` (`pipelines_lsst_io`_) has an EUPS table file.
+This table file defines all the Stack packages that should be set up and that are available for inclusion in the Stack documentation build.
+See :doc:`add-a-package-to-pipelines-lsst-io` for details.
+
+.. _stack-docs-system-build-linking:
+
+Package documentation linking
+-----------------------------
+
+The next step of the build is to run `stack-docs`_, from Documenteer_.
+
+The `stack-docs`_ app begins by discovering packages that are set up by EUPS and that also have :ref:`doc/manifest.yaml <docdir-manifest-yaml>` files.
+Following the :ref:`doc/manifest.yaml <docdir-manifest-yaml>` file, `stack-docs`_ symlinks the :ref:`module <docdir-module-doc-directories>` and :ref:`package documentation directories <docdir-package-doc-directory>` into the :file:`modules/` and :file:`packages/` directories of the `pipelines_lsst_io`_ repository.
+
+.. _stack-docs-system-build-sphinx:
+
+The Sphinx build
+----------------
+
+Next, `stack-docs`_ runs the Sphinx_ build on the `pipelines_lsst_io`_ repository.
+This is different from most projects that use a :file:`Makefile` and Sphinx_\ ’s builtin :command:`sphinx-build` app.
```

Comment:
This begs the question of why we need to be different.  That doesn't seem to be described, even in the documenteer docs.

Reply:
We have the `stack-docs build` command so do the symlinking in addition to a build and the `stack-docs clean` command removes additional intermediate build directories. I'll see if I can improve the documentation content in documenteer to describe this better.

We're not using Makefiles here because in the past invoking sphinx-build from a Makefile within the Stack's Python environment ran into SIP issues. I can try again now that we're installing into a virtual environment.